crates/core/src/pattern_compiler/compiler.rs

```diff
@@ -385,67 +382,70 @@ fn get_definitions(
     );
 
     for (file, pattern) in libs.iter() {
-        let context = CompilationContext {
-            src: pattern,
-            file,
-            ..*context
+        let mut node_context = NodeCompilationContext {
+            compilation: &CompilationContext {
+                src: pattern,
+                file,
+                ..*context
+            },
+            // This is the global scope, but we cannot pass `global_vars` due to aliasing.
+            // Thankfully, we don't really have global variables, so this map is never accessed.
+            vars: &mut BTreeMap::new(),
```

Comment:
It’s simply that previously the definition compilers wouldn’t receive any vars, but because they all follow the same signature now I have to pass an empty map.

Reply:
I see `This is the global scope` is kind of ambiguous here. This refers to the fact that we are not yet within the scope of a pattern definition. I would rephrase to say that there isn't yet a local pattern scope. There is a global scope, it's passed in 4 lines down on line 396.

Reply:
I've reworded it.